### PR TITLE
feat: APPS-2959 FlexibleMediaGallery FTVA variant

### PR DESCRIPTION
**Pull request by tinuola**
Connected to [APPS-2959](https://jira.library.ucla.edu/browse/APPS-2959)

**Components Updated:**
- MediaBadge.vue
- ~/Flexible/MediaGallery.vue
- ~/Flexible/MediaGallery/BannerImage.vue
- ~/Flexible/MediaGallery/NewLightbox.vue
- ~/Flexible/MediaGallery/ThumbnailCard.vue

**Stories:**
- ~/stories/MediaBadge.stories.js
- ~/stories/MediaGallery_BannerImage.stories.js
- ~/stories/MediaGallery_NewLightbox.stories.js
- ~/stories/MediaGallery_ThumbnailCard.stories.js
- ~/stories/Flexible_MediaGallery.stories.js

<br>

**Notes:**
- Updated multiple components that comprise the FlexibleMediaGallery with FTVA variants
- Added new prop `inline` to be used to set the Lightbox component to display as an inline component, preserving the default overlay behavior when needed
  - Any FTVA template currently using the Lightbox component inline should be updated with the prop 
  - Related: https://uclalibrary.atlassian.net/browse/APPS-2998
- Moved and updated CSS files
- Although not specified explicitly in the ticket or the design comps, I included a `halfWidth` gallery option/story for FTVA
  - If the gallery might be used on the FTVA site, it can be updated later with confirmed/theme specs

<br>

**StoryBook Previews:**
- MediaBadge
  - https://deploy-preview-626--ucla-library-storybook.netlify.app/?path=/story/media-badge--ftva

- Media Gallery BannerImage
  - https://deploy-preview-626--ucla-library-storybook.netlify.app/?path=/story/media-gallery-banner-image--ftva

- Media Gallery ThumbnailCard
  - https://deploy-preview-626--ucla-library-storybook.netlify.app/?path=/story/media-gallery-thumbnail-card--ftva

- NewLightbox (Default)
  - https://deploy-preview-626--ucla-library-storybook.netlify.app/?path=/story/media-gallery-new-lightbox--ftva-default-lightbox
  - Using `inline` prop set to `true`:
    - https://deploy-preview-626--ucla-library-storybook.netlify.app/?path=/story/media-gallery-new-lightbox--ftva-event-detail-carousel

- FlexibleMediaGallery (FullWidth)
  - https://deploy-preview-626--ucla-library-storybook.netlify.app/?path=/story/flexible-media-gallery--ftva

- FlexibleMediaGallery (HalfWidth)
  - https://deploy-preview-626--ucla-library-storybook.netlify.app/?path=/story/flexible-media-gallery--ftva-half-width

<br>

**Local Nuxt Previews (with craft data):**
- Collapsed/Default
media-gallery1

- Expanded
media-gallery2

- Lightbox
media-gallery3

<br>

**Checklist:**

-   [x] I checked that it is working locally in the dev server
-   [x] I checked that it is working locally in the storybook
-   [x] I checked that it is working locally in the 
library-website-nuxt dev server
-   [x] I added a screenshot of it working
-   [x] UX has reviewed and approved this
-   [x] I have requested a PR review from the dev team
-   [x] I used a conventional commit message
-   [x] I assigned myself to this PR

[APPS-2959]: https://uclalibrary.atlassian.net/browse/APPS-2959?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by axamei** [comment]
@tinuola the blue icon overlay should be a circle instead of the molecule. that is the only discrepancy with the design that i see

File: src/lib-components/Flexible/MediaGallery.vue [hunk 1]
```diff
@@ -54,20 +78,32 @@ function selectItem(itemIndex: number) {
...
       v-if="showLightboxModal"
       :items="block.mediaGallery"
       :selected-item="selectionIndex"
       tabindex="0"
+      :inline="inline"
       @close-modal="hideLightboxModal"
       @keydown.esc="hideLightboxModal"
     />
 
+    <section v-if="theme && fullWidthGallery" class="gallery-header">
```
Comment (farosFreed) [inline src/lib-components/Flexible/MediaGallery.vue:95]: can we be more explicit here and specify theme.value = ftva? Especially if there are ftva styles that this depends on

File: src/lib-components/Flexible/MediaGallery/BannerImage.vue [hunk 2]
```diff
@@ -72,9 +77,18 @@ const classes = computed(() => {
         />
       </div>
       <MediaBadge v-if="nItems > 1" :is-expanded="expanded">
-        {{ nItems }}
-        images
-        <svg class="glyph-expand">
+        <span v-if="theme">
```
Comment (farosFreed) [inline src/lib-components/Flexible/MediaGallery/BannerImage.vue:80]: same comment as before, can we specify theme.value === ftva? Or add a class to this space and use display:none to hide it on the default.scss theme file

File: src/lib-components/Flexible/MediaGallery/BannerImage.vue [hunk 3]
```diff
@@ -72,9 +77,18 @@ const classes = computed(() => {
...
+          <span v-if="expanded">Collapse Gallery</span>
+          <span v-else>View all {{ nItems }} images</span>
+        </span>
+        <span v-else>
+          {{ nItems }}
+          images
+        </span>
+
+        <!-- Toggle Icons -->
+        <SvgIconFtvaDropTriangle v-if="theme" class="drop-triangle" :class="{ 'is-expanded': expanded }" />
```
Comment (farosFreed) [inline src/lib-components/Flexible/MediaGallery/BannerImage.vue:90]: one more, can we specify theme.value === 'ftva'?

**Review by farosFreed — CHANGES_REQUESTED** [review]

**Review by pghorpade — APPROVED** [review]
this is comprehensive and well thought out PR. 💯

**Pull request merged**